### PR TITLE
Fix Bytes deserialization for unicode values

### PR DESCRIPTION
**Pull request by popzxc**
Current `Bytes` deserialization logic can panic if supplied non-ASCII values: it uses [`str::len`](https://doc.rust-lang.org/std/primitive.str.html#method.len) and [`SliceIndex<str>`](https://doc.rust-lang.org/std/primitive.str.html#impl-SliceIndex%3Cstr%3E-for-Range%3Cusize%3E), both of which are byte-based (not char-based).
The deserializer code can panic when `[0..2]` range is not on the char boundary.

This PR fixed it by checking whether the string is ASCII first.

### CODE REVIEW

File: src/types/bytes.rs [hunk 1]
```diff
@@ -54,7 +54,7 @@ impl<'a> Visitor<'a> for BytesVisitor {
     where
         E: Error,
     {
-        if value.len() >= 2 && &value[0..2] == "0x" {
+        if value.is_ascii() && value.len() >= 2 && &value[0..2] == "0x" {
```
Comment (tomusdrw) [inline src/types/bytes.rs:57]:
```suggestion
        if value.starts_with("0x") {
```

Wouldn't actually that work better?

**Review by tomusdrw — COMMENTED** [review]
Good catch! I proposed a change to maybe just use `starts_with`, can you check if it passes all the tests?

File: src/types/bytes.rs (hunk 1 above)
Comment (popzxc) [reply src/types/bytes.rs]: Oh, indeed! However, it can be expressed even better with `str::strip_prefix`. Pushed the change.

**Review by tomusdrw — APPROVED** [review]

**Comment by popzxc** [comment]
@tomusdrw I kind of need this fix for my project, could you please release a new version?

**Review by Armss9936 — APPROVED** [review]

**Pull request merged**